### PR TITLE
[gas] initial gas calibration implementation

### PR DESCRIPTION
**Pull request by vgao1996**
This introduces a script that would allow us to benchmark the Move VM using valgrind, which would give us a pretty accurate estimation of the number of machine instructions a certain part of our pipeline (be it Move or Rust) would cost, which we could use for gas metering purposes.

Right now this only covers the bytecode instructions and maybe deserialization and bytecode verification, but more will come in the future.

Note: this change is totally safe to check in as it doesn't touch any of the production code.

This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/3794)

### CODE REVIEW

File: .github/workflows/run-gas-calibration.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    runs-on: high-perf-docker
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # get all the history because cargo xtest --change-since origin/main requires it.
+      - uses: ./.github/actions/rust-setup
+      - name: install Valgrind
+        shell: bash
+        run: sudo apt-get -y install valgrind
+      - run: cargo run -p aptos-gas-calibration -- -r
```
Comment (davidiw) [inline .github/workflows/run-gas-calibration.yaml:35]: is the intent to provide some sort of test?

**Review by davidiw — COMMENTED** [review]
seems good for me... would be great to have a readme

**Comment by davidiw** [comment]
@vgao1996 can we get a readme here, and then we can land this?

**Comment by vgao1996** [comment]
@davidiw Sure I'll get to this sometime. It's a lower priority compared to some other tasks though...

**Comment by davidiw** [comment]
@vgao1996 ☝️

**Comment by gregnazario** [comment]
@vgao1996 is this still happening?  Otherwise, should we close this PR?

**Comment by vgao1996** [comment]
Let's still check the tool in. It's better than having nothing.

**Review by Moonsixtus14 — APPROVED** [review]

**Review by movekevin — APPROVED** [review]

**Review by georgemitenkov — APPROVED** [review]

**Pull request merged**